api/src/main/java/org/eclipse/microprofile/faulttolerance/Fallback.java

```diff
@@ -44,4 +44,10 @@
      * @return the fallback class
      */
     Class<? extends FallbackHandler<?>> value();
+    /**
+     * Specify the method name to be fallbacked to. This method belongs
+     * to the same class as the method to fallback.
+     * @return the local method to fallback to
+     */
+    String fallbackMethod() default "";
```

Comment:
If FallbackHandler is specified, it will have only one method handle(ExecutionContext context). There will be any method called the name specified by fallbackMethod.
In light of your concern, we can say :
-  If FallbackHandler is specified, then use handle on the handler to execute the fallback 
-  If no handler specified, then invoke the method specified by the fallbackMethod on the same class
- If both specified, try to fallback to fallbackMethod first, if it fails then call FallbackHandler.handle. or we just thrown DeploymentException?
By the way, for validating fallbackMethod, its argument could be either no arg or exactly same as the original method.